src/utils/SubscribableRequestQueue.ts

```diff
@@ -0,0 +1,175 @@
+import RequestQueue from "./RequestQueue";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Resolver = (value?: any) => void;
+type Rejecter = (reason?: unknown) => void;
+
+type RequestSubscription = {
+  subscriberId: number;
+  resolve: Resolver;
+  reject: Rejecter;
+};
+
+/**
+ * An extension of `RequestQueue` that adds a concept of "subscribers," which may share references to a single request
+ * or cancel their subscription without disrupting the request for other subscribers.
+ */
+export default class SubscribableRequestQueue {
+  private queue: RequestQueue;
+
+  /** Number of subscribers, used for generating unique subscriber IDs. */
+  private numSubscribers: number;
+  /** Map keyed by subscriber ID. Subscribers are only useful for cancelling early, so we only store rejecters here. */
```

Comment:
the comment doesn't make it clear what the Map<string, Rejecter> is for. We're keying by subscriberid but what is the value?